### PR TITLE
[Merged by Bors] - chore: rename `mkpair` → `pair`

### PR DESCRIPTION
**Pull request by thorimur**
We rename the ported `mkpair` to `pair`, as `mkpair` didn't follow the naming convention and `pair` pairs well with `unpair`. A poll for this decision was held on [zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/naming.20bikeshed.3A.20.60mkpair.60.2F.60unpair.60).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Nat/Pairing.lean [hunk 1]
```diff
@@ -50,23 +50,23 @@ def unpair (n : ℕ) : ℕ × ℕ :=
 #align nat.unpair Nat.unpair
 
 @[simp]
-theorem mkpair_unpair (n : ℕ) : mkpair (unpair n).1 (unpair n).2 = n := by
+theorem mkPair_unpair (n : ℕ) : mkPair (unpair n).1 (unpair n).2 = n := by
```
Comment (eric-wieser) [inline Mathlib/Data/Nat/Pairing.lean:53]: I wonder if the motivation here was to be consistent with `unpair`? Is `mkPair` / `unPair` better or worse than this PR / what we had before?

**Comment by ChrisHughes24** [comment]
bors merge

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**